File: java/hamming/src/main/java/Hamming.java
public class Hamming {
	public static int compute(String a, String b){
		int dif = 0;
		if(a.length() != b.length()) throw new java.lang.IllegalArgumentException();
		for(int i = 0; i < a.length(); i++)
			dif += a.charAt(i) != b.charAt(i) ? 1 : 0;
		return dif;
	}
}

File: ruby/hamming/hamming.rb
class Hamming
	VERSION = 1

	def self.compute(a, b)
		if a.length != b.length then
			raise ArgumentError, "msg", caller
		end
		dif = 0
		for i in 0..a.length - 1
			if a[i].chr != b[i].chr then
				dif = dif + 1
			end
		end
		return dif
	end
end
